src/app/games/yinsh/yinsh.component.ts

```diff
@@ -353,15 +352,14 @@ export class YinshComponent
             this.constructedState.sideRings,
             this.constructedState.turn);
 
-        switch (this.movePhase) {
-            case 'INITIAL_CAPTURE_SELECT_FIRST':
-            case 'INITIAL_CAPTURE_SELECT_LAST':
-                this.movePhase = 'INITIAL_CAPTURE_SELECT_RING';
-                break;
-            case 'FINAL_CAPTURE_SELECT_FIRST':
-            case 'FINAL_CAPTURE_SELECT_LAST':
-                this.movePhase = 'FINAL_CAPTURE_SELECT_RING';
-                break;
+        if (this.movePhase === 'INITIAL_CAPTURE_SELECT_FIRST' ||
+            this.movePhase === 'INITIAL_CAPTURE_SELECT_LAST')
+        {
+            this.movePhase = 'INITIAL_CAPTURE_SELECT_RING';
+        } else {
+            const message: string = 'selectCapture did not expect to be called in movePhase ' + this.movePhase;
+            assert(this.movePhase === 'FINAL_CAPTURE_SELECT_FIRST' || this.movePhase === 'FINAL_CAPTURE_SELECT_LAST', message);
```

Comment:
Utils.expectToBe

Reply:
Mais les expectToBe on les utilise que pour les default des switch, pas pour tout les assert dont la condition est un === si :D ? (pour éviter le "voir commentaire précédént/suivant/ailleurs" 🗡️

Reply:
voir la réponse précédente :dagger:

Reply:
mais, y a un or, ça va moins le faire pour expected to be.